### PR TITLE
Fail when unexpected version of kotlin-compiler-embeddable is on runtime classpath

### LINKED ISSUE #5551: detekt 1.22.0 no longer works with kotlin <1.7
## Expected Behavior

detekt 1.22.0 continues to work with projects using kotlin 1.6.21, as the doc doesn't mention kotlin 1.7 as a requirement.

## Observed Behavior

```
Caused by: java.lang.reflect.InvocationTargetException
        at jdk.internal.reflect.GeneratedMethodAccessor2658.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at io.gitlab.arturbosch.detekt.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:58)
        ... 125 more
Caused by: java.lang.NoSuchMethodError: 'void org.jetbrains.kotlin.cli.jvm.config.JvmContentRootsKt.configureJdkClasspathRoots(org.jetbrains.kotlin.config.CompilerConfiguration)'
        at io.github.detekt.parser.KotlinEnvironmentUtilsKt.createCompilerConfiguration(KotlinEnvironmentUtils.kt:122)
        at io.gitlab.arturbosch.detekt.core.settings.EnvironmentFacade$environment$2.invoke(EnvironmentAware.kt:36)
        at io.gitlab.arturbosch.detekt.core.settings.EnvironmentFacade$environment$2.invoke(EnvironmentAware.kt:35)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at io.gitlab.arturbosch.detekt.core.settings.EnvironmentFacade.getEnvironment(EnvironmentAware.kt:35)
        at io.gitlab.arturbosch.detekt.core.ProcessingSettings.getEnvironment(ProcessingSettings.kt)
        at io.gitlab.arturbosch.detekt.core.KtTreeCompiler.<init>(KtTreeCompiler.kt:14)
        at io.gitlab.arturbosch.detekt.core.tooling.ParsingStrategyKt$inputPathsToKtFiles$1.invoke(ParsingStrategy.kt:24)
        at io.gitlab.arturbosch.detekt.core.tooling.ParsingStrategyKt$inputPathsToKtFiles$1.invoke(ParsingStrategy.kt:23)
        at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$analyze$filesToAnalyze$1.invoke(Lifecycle.kt:38)
        at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$analyze$filesToAnalyze$1.invoke(Lifecycle.kt:38)
        at io.gitlab.arturbosch.detekt.core.util.PerformanceMonitor.measure(PerformanceMonitor.kt:42)
        at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.measure(Lifecycle.kt:34)
        at io.gitlab.arturbosch.detekt.core.tooling.Lifecycle$DefaultImpls.analyze(Lifecycle.kt:38)
        at io.gitlab.arturbosch.detekt.core.tooling.DefaultLifecycle.analyze(Lifecycle.kt:61)
        at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade$runAnalysis$1.invoke(AnalysisFacade.kt:48)
        at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade$runAnalysis$1.invoke(AnalysisFacade.kt:47)
        at io.gitlab.arturbosch.detekt.core.tooling.ProcessingSpecSettingsBridgeKt.withSettings(ProcessingSpecSettingsBridge.kt:26)
        at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.runAnalysis$detekt_core(AnalysisFacade.kt:47)
        at io.gitlab.arturbosch.detekt.core.tooling.AnalysisFacade.run(AnalysisFacade.kt:25)
        at io.gitlab.arturbosch.detekt.cli.runners.Runner.call(Runner.kt:33)
        at io.gitlab.arturbosch.detekt.cli.runners.Runner.execute(Runner.kt:23)
        ... 128 more
```

## Context

detekt calls https://cs.github.com/JetBrains/kotlin/blob/b6481ed891a35d5f148209a9666462b5248046e3/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/config/JvmContentRoots.kt#L91 which does not seem to exist in kotlin 1.6: https://github.com/JetBrains/kotlin/blob/1.6.20/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/config/JvmContentRoots.kt

## Your Environment
* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.4.2
* Version of kotlin compiler: 1.6.21

### LINKED ISSUE #5645: Report or fail build when `kotlin-compiler-embeddable` version is overridden
## Expected Behavior
detekt runs successfully using the expected version of the `kotlin-compiler-embeddable` dependency. If the version of the dependency is overridden in the user's build, either warn the user or fail the build, with a link to documentation explaining the issue and some ways to deal with it.

## Current Behavior
This keeps coming up as an issue that detekt doesn't properly handle.

## Context
#5644 #5582 #5551 #5021 #4786 #4287

### PR DESCRIPTION
**Pull request by 3flex**
Fixes #5645 
Closes #5551

### CODE REVIEW

File: website/docs/gettingstarted/gradle.mdx [hunk 1]
```diff
@@ -490,3 +490,31 @@ Instead of disabling detekt for the check task, you may want to increase the bui
 detekt comes with an [IntelliJ Plugin](https://plugins.jetbrains.com/plugin/10761-detekt) that you can install directly from the IDE. The plugin offers warning highlight directly inside the IDE as well as support for code formatting.
 
 The source code of the plugin is available here: [detekt/detekt-intellij-plugin](https://github.com/detekt/detekt-intellij-plugin)
+
+## <a name="dependencies">Gradle runtime dependencies</a>
+
+detekt is tightly coupled to the Kotlin compiler and requires a specific version to be available at runtime to perform
```
Comment (atulgpt) [inline website/docs/gettingstarted/gradle.mdx:496]: Hi @3flex should this be compile time or runtime? As detekt will run in any project at compile time. Right?
BTW I am no expert in class paths so I might be wrong 😅

Comment (3flex) [reply website/docs/gettingstarted/gradle.mdx:496]: Yeah that might be confusing, it's not the project's runtime or compile classpath, but the classpath detekt is run with, which is different. I'm trying to keep the description succinct but I'm open to suggestions.

Comment (atulgpt) [reply website/docs/gettingstarted/gradle.mdx:496]: I think saying `build runtime to perform` will make it clear that it is referring to classpath of gradle build

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt [hunk 2]
```diff
@@ -50,6 +52,12 @@ fun buildRunner(
     outputPrinter: PrintStream,
     errorPrinter: PrintStream
 ): Executable {
+    check(KotlinCompilerVersion.VERSION == whichKotlin()) {
```
Comment (atulgpt) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt:55]: here are we allowing both values to be `null`? As `whichKotlin` can theoretically return `null` and the same with `KotlinCompilerVersion.VERSION` which can return `null` if there is an IO error reading the `compiler.version` file. I think we can also disallow the null value

File: website/docs/gettingstarted/gradle.mdx (hunk 1 above)
Comment (3flex) [reply website/docs/gettingstarted/gradle.mdx:496]: Updated - hopefully this is a little clearer.

Comment (atulgpt) [reply website/docs/gettingstarted/gradle.mdx:496]: Yup now to me it's more clearer

**Review by schalkms — APPROVED** [review]
👍

**Pull request merged**